x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/Negateable.java

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.expression.function.scalar;
+
+public interface Negateable {
+
```

Comment:
No need to extract the interface so high up - just put it inside the predicate package since essentially it applies only to logical predicates. Also the correct name should be `Negatable`. 

Also it's worth looking into adding the generic aspect in place to have a better lock down on the class hierarchy:
`Negatable<T extends ScalarFunction> { T negate() }` . IsNull/IsNotNull can lock it to `UnaryScalarFunction` while the `BinaryLogic` keep the `BinaryOperator` in place.

Reply:
But i didn't change its name: `Negate` + `able` = `Negateable`